R/format_data.R

```diff
@@ -1467,24 +1470,26 @@ fmt_date <- function(data,
     fns = list(
       default = function(x) {
 
-        # If `x` is of the `Date` type, simply make
-        # that a character vector
-        if (inherits(x, "Date")) {
+        # If `x` is of the `Date` or `POSIXct` types, simply transform
+        # `x` into a character vector
+        if (inherits(x, "Date") || inherits(x, "POSIXct")) {
           x <- as.character(x)
         }
 
-        date <-
-          ifelse(grepl("^[0-9]*?\\:[0-9]*?", x), paste("1970-01-01", x), x) %>%
-          strftime(format = date_format_str)
+        # If the incoming string is a time then append the `1970-01-01` date
+        date <- ifelse(grepl("^[0-9]*?\\:[0-9]*?", x), paste("1970-01-01", x), x)
+
+        # Format the date string using `strftime()`
+        date <- strftime(date, format = date_format_str)
 
-        if (date_style %in% 2:12) {
-          date <- date %>% tidy_gsub(., "^0", "")
+        # Perform several cosmetic changes to the formatted date
+        if (date_format_str != "%F") {
```

Comment:
If you accept custom strftime format strings, then don't do this

Reply:
This codepath is avoided when providing a `format` in `fmt_datetime()`.